lib/matplotlib/testing/exceptions.py

```diff
@@ -1,15 +1,3 @@
-class KnownFailureTest(Exception):
```

Comment:
I cannot agree with this for two reasons:
  - PRs that relay on `raise KnownFailureTest` should not pass after this PR ever, and this change will force them to switch on `xfail`
  - Other projects must not relay on matplotlib testing features, it is their problem if they are

I can understand backward compatibility on matplotlib core features, but with removing nose support all this stuff will gone. I have already tried my best to make dual nose-pytest support, but leaving nose-specific stuff on the current places will blow something in any time in the future.